app/components/Authentication/Github/Github.jsx

```diff
@@ -1,248 +1,513 @@
+            })
     }
 
-    requestOrganization() {
-        this.setState({
-            requestOrganization: this.state.organization ? false : true
-        })
+    listGithubUsers() {
+        tokenHasCapabilities(['list'], `${this.state.baseVaultPath}/map/users`)
+            .then(() => {
+                callVaultApi('get', `${this.state.baseVaultPath}/map/users`, { list: true }, null)
+                    .then((resp) => {
+                        let users = resp.data.data.keys;
+                        this.setState({ users: _.valuesIn(users) });
+                    })
+                    .catch((error) => {
+                        if (error.response.status !== 404) {
+                            snackBarMessage(error);
+                        } else {
+                            this.setState({ users: [] });
+                        }
+                    });
+            })
+            .catch(() => {
+                snackBarMessage(new Error("Access denied"));
+            })
     }
 
-    renderOrganizationDialog() {
-        const actions = [
-            <div>
-                <FlatButton label="Close" primary={true} onTouchTap={(e) => closeDialog(e)} />
-                <FlatButton label="Submit" secondary={true} onTouchTap={(e) => submitOrg(e)} />
-            </div>
-        ];
-
-        let submitOrg = (e) => {
-            window.localStorage.setItem('githubOrganization', this.state.tmpOrganization)
-            this.setState({
-                organization: this.state.tmpOrganization,
-                requestOrganization: false
-            });
-        };
-
-        let closeDialog = (e) => {
-            this.setState({
-                requestOrganization: false
-            });
-        };
-
-        return (
-            <Dialog
-                title="Organization"
-                actions={actions}
-                modal={true}
-                open={this.state.requestOrganization}
-            >
-                <TextField
-                    id="organization"
-                    fullWidth={true}
-                    className="col-xs-12"
-                    defaultValue={this.state.organization}
-                    onChange={(e, v) => this.setState({ tmpOrganization: v })}
-                />
-                <div className={styles.error}>{this.state.errorMessage}</div>
-            </Dialog>
-        )
+    getOrgConfig() {
+        tokenHasCapabilities(['read'], `${this.state.baseVaultPath}/config`)
+            .then(() => {
+                callVaultApi('get', `${this.state.baseVaultPath}/config`, null, null)
+                    .then((resp) => {
+                        let config = resp.data.data;
+                        this.setState({
+                            config: update(this.state.config,
+                                {
+                                    organization: { $set: (config.organization ? config.organization : undefined) },
```

Comment:
why do these even need to be switched at all? if config exists should be the only check prior. Otherwise it will be set to undefined no matter what.